spec/operators/catch-spec.ts

```diff
@@ -278,4 +279,84 @@ describe('Observable.prototype.catch', () => {
       done();
     });
   });
+
+  context('fromPromise', () => {
+    type SetTimeout = (callback: (...args: any[]) => void, ms: number, ...args: any[]) => NodeJS.Timer;
+
+    let trueSetTimeout: SetTimeout;
+    let sandbox: sinon.SinonSandbox;
+    let timers: sinon.SinonFakeTimers;
+
+    let source: Rx.Observable<any>;
+    beforeEach(() => {
```

Comment:
These tests are a little too DRY for my liking. Prefer doing as much set up as possible in each test.

Also, I'll be honest, I'm not at all familar with sinon.sandbox. I'll need to research it to see if it's an approach we want to use. I presume it's some virtualized scheduling thing.